tools/bamm-cli/src/main/java/io/openmanufacturing/sds/BammCli.java

```diff
@@ -12,542 +12,52 @@
-import io.vavr.CheckedFunction1;
-import io.vavr.control.Option;
-import io.vavr.control.Try;
-
+import org.fusesource.jansi.AnsiConsole;
+
+import io.openmanufacturing.sds.aspect.AspectCommand;
+import picocli.CommandLine;
+
+@CommandLine.Command( name = BammCli.COMMAND_NAME,
+      description = "Command line tool for working with Aspect Models",
+      subcommands = { CommandLine.HelpCommand.class },
+      headerHeading = "@|bold Usage|@:%n%n",
+      descriptionHeading = "%n@|bold Description|@:%n%n",
+      parameterListHeading = "%n@|bold Parameters|@:%n",
+      optionListHeading = "%n@|bold Options|@:%n",
+      footer = "%nRun @|bold " + BammCli.COMMAND_NAME + " help <command>|@ to display its help."
+            + "%nDocumentation: https://openmanufacturingplatform.github.io/sds-documentation/"
+)
 @SuppressWarnings( "squid:S1147" ) // System.exit is really required here, this is a CLI tool
-public class BammCli {
-   private static final Logger LOG = LoggerFactory.getLogger( BammCli.class );
-   private static final ObjectMapper objectMapper = new ObjectMapper();
-
-   private static void fileNotFound( final Args args, final Throwable exception ) {
-      LOG.error( "Can not open file for reading: {}", args.input, exception );
-      System.exit( 1 );
-   }
-
-   private static FileOutputStream getStreamForFile( final String path, final String artifactName, final Args args ) {
-      try {
-         final File directory = new File( args.outputPath.getPath() + File.separator + path );
-         directory.mkdirs();
-         final File file = new File( directory.getPath() + File.separator + artifactName );
-         return new FileOutputStream( file );
-      } catch ( final FileNotFoundException exception ) {
-         fileNotFound( args, exception );
-      }
-      return null;
-   }
-
-   private static FileOutputStream getStreamForFile( final String artifactName, final Args args ) {
-      return getStreamForFile( "", artifactName, args );
-   }
-
-   private static void printHelp( final JCommander jCommander ) {
-      final Properties properties = new Properties();
-      Stream.of( "application", "git" )
-            .map( fileName -> BammCli.class.getResourceAsStream( "/" + fileName + ".properties" ) )
-            .forEach( inputStream -> {
-               try {
-                  properties.load( inputStream );
-                  inputStream.close();
-               } catch ( final IOException exception ) {
-                  LOG.error( "Error while closing input stream", exception );
-               }
-            } );
+public class BammCli extends AbstractCommand {
 
-      final String name = properties.getProperty( "application.name" );
-      final String version = properties.getProperty( "version" );
-      final String buildDate = properties.getProperty( "build.date" );
-      LOG.warn( "{}\n  Version: {}\n  Build date: {}", name, version, buildDate );
+   public static final String COMMAND_NAME = "bamm-cli";
```

Comment:
I would actually change this to just "bamm", and also update the sample commands in the documentation accordingly. Just like Azure CLI's command is just "az" etc., we should not include "cli" in the command name.